docs/non-normative/code-generation.md

```diff
@@ -0,0 +1,187 @@
+Experimental semantic conventions are intended for end-user applications.
+
+### Deprecated Conventions
+
+It's recommended to generate code for deprecated attributes, metrics, and other conventions. Use appropriate annotations to mark them as deprecated.
+Conventions have a `stability` property which provide the stability level at the deprecation time (`experimental` or `stable`) and
+the `deprecated` property that describes deprecation reason which can be used to generate documentation.
+
+- Deprecated conventions that reached stability should not be removed without major version update according to SemVer.
+- Conventions that were deprecated while being experimental should still be generated and kept in the preview (part of) semantic conventions artifact. It minimizes runtime issues
+  and breaking changes in user applications.
+
+Keep stable convention definitions inside the preview (part of) semantic conversions artifact. It prevents user code from breaking when semantic convention stabilizes. Deprecate stable definitions inside the preview artifact and point users to the stable location in generated documentation.
+For example, in Java `http.request.method` attribute is defined as the deprecated `io.opentelemetry.semconv.incubating.HttpIncubatingAttributes.HTTP_REQUEST_METHOD` field and also as stable `io.opentelemetry.semconv.HttpAttributes.HTTP_REQUEST_METHOD`.
+
+## Semantic Conventions Artifact Structure
+
+This section contains suggestions on structuring semantic convention artifact(s) which should be adjusted to the specific language.
+
+- Artifact name:
+  - `opentelemetry-semconv` - stable conventions
+  - `opentelemetry-semconv-incubating` - (if applicable) the preview artifact containing all conventions
+- Namespace: `opentelemetry.semconv` and `opentelemetry.semconv.incubating`
+- All supported Schema URLs should be listed to allow different instrumentations in the same application to provide the exact version of conventions they follow.
+- Attributes, metrics, and other convention definitions should be grouped by the convention type and the root namespace. See the example below:
+
+```
+├── SchemaUrls.code
+├── attributes
+│   ├── ClientAttributes.code
+│   ├── HttpAttributes.code
+│   └── ...
+├── metrics
+│   ├── HttpMetrics.code
+│   └── ...
+└── events
+    └── ...
+```
+
+## Generating semantic conventions
+
+This section describes how to do code-generation with weaver.
+
+> Note:
+> We're transitioning from [build-tools](https://github.com/open-telemetry/build-tools/blob/main/semantic-conventions/README.md#code-generator)
+> to [opentelemetry-weaver](https://github.com/open-telemetry/weaver/blob/main/crates/weaver_forge/README.md) to generate code for semantic conventions.
+> All new code-generation should be done using weaver, build-tools may become incompatible with future version of semantic conventions.
+
+Code-generation is based on YAML definitions in the specific version of semantic conventions and usually
```

Comment:
```suggestion
Code-generation is based on YAML definitions in the specific version of semantic conventions.
Usually, it involves several steps where some can be semi-automated:
```

To avoid the sentence to be too long :)